assignment3.txt

```diff
@@ -0,0 +1,21 @@
+1. Is the following code valid? If so, what information or assumptions must be given for it to be 
+valid? If not, why not?
+	for (; i < 1000; i*= 2) {
+		printf("%d\n");
+	}
+The code is not valid for a few reasons. One is that there is no main statement or include statements
+but I will assume that those are assumed to be correct and focus on this fragment. In this fragment,
+the variable i is never declared or assigned a value. It could be either float or int at this point.
+In the case that it was int, the printf statement would be correct with %d, however after the
+quote statement it would need a comma and then the name of the variable.
```

Comment:
perfect! It's also important to note that the i would need to be initialized to something besides 0 as to not enter an infinite loop.